test/rules/yaml_test_cases_test.py

```diff
@@ -9,14 +9,21 @@
 )
 from sqlfluff.core.config import FluffConfig
 
-ids, test_cases = load_test_cases(
-    test_cases_path=os.path.join("test/fixtures/rules/std_rule_cases", "*.yml")
-)
+
+def pytest_generate_tests(metafunc):
+    """Generate tests, optionally by rule_id."""
+    rule_id = metafunc.config.getoption("rule_id")
+    ids, test_cases = load_test_cases(
+        test_cases_path=os.path.join(
+            "test/fixtures/rules/std_rule_cases", f"{rule_id}.yml"
+        )
+    )
+    if "test_case" in metafunc.fixturenames:
+        metafunc.parametrize("test_case", test_cases, ids=ids)
 
 
 @pytest.mark.integration
 @pytest.mark.rules_suite
-@pytest.mark.parametrize("test_case", test_cases, ids=ids)
 def test__rule_test_case(test_case, caplog):
```

Comment:
What happens with this test after removing `parametrize`?

BTW, in the past, I have used pytest's `-k` option to run the tests for one rule. It's a little less precise, because I think it just looks for a substring in the test name, but in practice, it has worked well.

Reply:
It stills runs all the combinations, see `By default, we still run everything:` part of the PR description.

Reply:
The CI still fails if one of the rule tests fails (reverted it back): https://github.com/sqlfluff/sqlfluff/pull/4963/commits/fbef8b0f22a459bb065ce363f86d51e57777a556

Reply:
Got it. Probably that ` metafunc.parametrize()` call. Fancy pytest stuff. I salute you!